cmd/frontend/graphqlbackend/search_results.go

```diff
@@ -1624,44 +1624,44 @@ func (r *searchResolver) determineRepos(ctx context.Context, tr *trace.Trace, st
 	return resolved, nil, nil
 }
 
-// Surface an alert if a query exceeds limits that we place on search. Currently limits
-// diff and commit searches where more than repoLimit repos need to be searched.
-func alertOnSearchLimit(resultTypes []string, args *search.TextParameters) ([]string, *searchAlert) {
-	limits := searchLimits()
-	// we don't need to handle the error here, because we don't have context and the
-	// promise can only return context errors
-	repos, _ := getRepos(context.Background(), args.RepoPromise)
+type DiffCommitError struct {
+	ResultType string
+	Max        int
+}
 
-	for _, resultType := range resultTypes {
-		if resultType != "commit" && resultType != "diff" {
-			continue
-		}
+type DiffCommitRepoLimitErr DiffCommitError
+type DiffCommitTimeLimitErr DiffCommitError
```

Comment:
```suggestion
type RepoLimitErr DiffCommitError
type TimeLimitErr DiffCommitError
```

I think we can be a little bit less verbose here. 

Also, at some point, naming like `DiffCommitError` should just be `CommitError` or `CommitSearchError`. Because the current distinction between `diff` and `commit` is imprecise to me--both search over commit data, they are just over different data associated with a commit, no? `diff` is the code change associated with a commit, and when we search for `type:commit` we generally mean "commit message". From the repo limit/time limit error point of view, a diff or commit-message search error are due to the same underlying limitation (repo limit).